### ISSUE TITLE
Join forces with Diagnostics WG

### ISSUE THREAD

**Issue by mhdawson** [nodejs/post-mortem#48]
The diagnostics wg (https://github.com/nodejs/Diagnostics) is restarting regular meetings and I think we share some common interests with that group.

I'd like to suggest that we join forces by using the same meeting to discuss post-mortem and other diagnostics issues.   The orgnanizer of the meeting and the group in the last meeting was open to this suggestion.

I've added the diag-agenda label to this repo and we'd just have to add that to any issues we wanted on the agenda.

I'm also planning to set up an in person Diagnostics Summit and I think it should cover both post-mortem and other Diagnostics topics.

@nodejs/post-mortem what do you think ?

@nodejs/diagnostics FYI

**Comment by cjihrig** [comment]
Makes sense to me.

**Comment by richardlau** [comment]
We used to do this -- nodejs/post-mortem members were cc'd and invited to the diagnostics WG meetings (e.g. https://github.com/nodejs/diagnostics/issues/79).

**Comment by Qard** [comment]
At the time post-mortem was starting the tracing working group had not yet expanded to cover general diagnostics, so it made sense for them to be separate. The groups probably should have merged when the switch to diagnostics happened though.

**Comment by mhdawson** [comment]
Ok seems like there is general agreement with the approach.  It would be great if the @nodejs/diagnostics team members could joing the next scheduled meeting for the Diagnostics group which is set for: 3-4 EST on Wed Nov 29th.

@mike-kaufman can you make sure to include any items tagged with the diagnostics-agenda Label are included in the agenda.

**Comment by richardlau** [comment]
I agree with @Qard that the groups should have been merged. If post-mortem needs to be kept as a team it should be under the diagnostics WG.

Any objections from @nodejs/post-mortem and/or @nodejs/diagnostics?

**Comment by mhdawson** [comment]
+1 from me.  I think we should look at migrating open issues over to the Diagnostics repo and making post-mortem a sub-team.

**Comment by mmarchini** [comment]
Are there any objections here? I want to start migrating open issues to [nodejs/diagnostics](https://github.com/nodejs/diagnostics) next week.

**Comment by joyeecheung** [comment]
+1

**Comment by mhdawson** [comment]
I think that this has been open long enough that we likely would have heard objections if there were going to be any.

There was agreement of all those at the diagnostic session at the summit last week as well.

@mmarchini I think if you don't hear any objections by next week you should be good to go.

**Comment by mmarchini** [comment]
Quick question: should I close issues here after migrating them to nodejs/diagnostics?

**Comment by mike-kaufman** [comment]
> should I close issues here after migrating them to nodejs/diagnostics?

Yeah, I think that makes sense - we ideally want a single copy of each issue where we concentrate discussion/decisions, vs having multiple copies.

**Comment by mhdawson** [comment]
+1 for closing here once there is something equivalent in the Diagnostics repo

**Comment by mmarchini** [comment]
@joyeecheung @mhdawson I don't have permission to close issues here :(

Shouldn't nodejs/post-mortem be a subteam of nodejs/diagnostics?

**Comment by richardlau** [comment]
> Shouldn't nodejs/post-mortem be a subteam of nodejs/diagnostics?

Ultimately yes (https://github.com/nodejs/post-mortem/issues/48#issuecomment-365456272). It's not at the moment because historically the two have been separate working groups (albeit with enough overlap between members that effectively all post-mortem work is being done under diagnostics and elsewhere in the org which is the main reason for folding post-mortem into diagnostics).

**Comment by richardlau** [comment]
I am actually wondering how many of the issues still open here need to be migrated given the lack of activity. Either the activity is happening elsewhere (in which case there should be another issue/repository elsewhere in the organization) or it has stalled (in which case the issue could be closed with an open invitation to open a new issue over in https://github.com/nodejs/diagnostics?).

**Comment by joyeecheung** [comment]
All the other open issues here are more than 1 year old,  and the discussions seem quite outdated now. If anyone wants to take another stab at them now, it's probably better to open a new issue as the context may have changed a lot since then. I think we should just close them and archive this repo, the previous discussions are useful for archeology but are unlikely to get any new traction as-is.

**Comment by richardlau** [comment]
I'm going through the remaining items and closing them with the following:

> Closing due to inactivity. If this work still needs to be tracked, please open a new issue over in https://github.com/nodejs/diagnostics.

**Comment by richardlau** [comment]
The only remaining open issue in this repository is this one 😁. I think the next steps are:
- [x] Open a PR in https://github.com/nodejs/TSC against https://github.com/nodejs/TSC/blob/master/WORKING_GROUPS.md that merges the Post-mortem WG charter into the Diagnostics WG charter (raised https://github.com/nodejs/TSC/pull/586)
- [x] Reorganisation of the GitHub nodejs/post-mortem team as a subteam of nodejs/diagnostics
- [x] Add a redirect to https://github.com/nodejs/diagnostics
- [ ] Update all references to this WG in repositories owned by this WG (https://github.com/nodejs/llnode, https://github.com/nodejs/node-report) to point to the equivalent in https://github.com/nodejs/diagnostics
- [ ] Archive this repository

**Comment by richardlau** [comment]
>  Add a redirect to https://github.com/nodejs/diagnostics

I've updated the description of this repository:
![image](https://user-images.githubusercontent.com/5445507/44161645-dfdb0e00-a0b5-11e8-9ad6-a14d7cd79cb6.png)

**Comment by joyeecheung** [comment]
I've put the post-mortem team under the diagnostics team. Should we do the same for https://github.com/orgs/nodejs/teams/post-mortem-admins ?

**Comment by richardlau** [comment]
> I've put the post-mortem team under the diagnostics team. Should we do the same for https://github.com/orgs/nodejs/teams/post-mortem-admins ?

I don't see why not. AIUI that team is used by our CI to control access to post-mortem jobs (the node-report one for definite, I can't remember if there were others and am currently locked out of the CI while the security releases are prepped).

**Comment by joyeecheung** [comment]
We may also want to do a review on the membership like https://github.com/nodejs/diagnostics/issues/217 and put the list of members to the README of the diagnostics repo

**Comment by richardlau** [comment]
That's a fair point. For example, I believe I am a member of this WG but not currently a member of the diagnostics WG.

**Comment by mhdawson** [comment]
I agree we don't want to just add people from this WG to the Diagnostics team.  I think we need to ask the existing members to confirm they are interested and will be active participants.  Since there has not been a post-mortem meeting in a long time I expect that the interested/active parties will mostly already be members of the Diagnostics WG.
